source/ranges.tex

```diff
@@ -6096,6 +6146,305 @@
+    constexpr V base() && { return std::move(@\exposid{base_}@); }
+
+    constexpr @\exposid{iterator}@ begin();
+
+    constexpr auto end() {
+      if constexpr (@\libconcept{common_range}@<V>) {
+        return @\exposid{iterator}@{*this, ranges::end(@\exposid{base_}@), {}};
+      } else {
+        return @\exposid{sentinel}@{*this};
+      }
+    }
+
+    constexpr subrange<iterator_t<V>> @\exposid{find-next}@(iterator_t<V>); // \expos
+  };
+
+  template<class R, class P>
+    split_view(R&&, P&&) -> split_view<views::all_t<R>, views::all_t<P>>;
+
+  template<@\libconcept{forward_range}@ R>
+    split_view(R&&, range_value_t<R>)
+      -> split_view<views::all_t<R>, single_view<range_value_t<R>>>;
+}
+\end{codeblock}
+
+\begin{itemdecl}
+constexpr split_view(V base, Pattern pattern);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+Initializes \exposid{base_} with \tcode{std::move(base)}, and
+\exposid{pattern_} with \tcode{std::move(pattern)}.
+\end{itemdescr}
+
+\begin{itemdecl}
+template<@\libconcept{forward_range}@ R>
+  requires @\libconcept{constructible_from}@<V, views::all_t<R>> &&
+           @\libconcept{constructible_from}@<Pattern, single_view<range_value_t<R>>>
+constexpr split_view(R&& r, range_value_t<R> e);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+Initializes \exposid{base_} with \tcode{views::all(std::forward<R>(r))}, and
+\exposid{pattern_} with \tcode{single_view\{std::move(e)\}}.
+\end{itemdescr}
+
+\begin{itemdecl}
+constexpr @\exposid{iterator}@ begin();
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\returns
+\tcode{\{*this, ranges::begin(\exposid{base_}), \exposid{find-next}(ranges::begin(\exposid{base_}))\}}.
+
+\pnum
+\remarks
+In order to provide the amortized constant time complexity
+required by the \libconcept{range} concept,
+this function caches the result within the \tcode{split_view}
+for use on subsequent calls.
+\end{itemdescr}
+
+\begin{itemdecl}
+constexpr subrange<iterator_t<V>> @\exposid{find-next}@(iterator_t<V> it); // \expos
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+Equivalent to:
+\begin{codeblock}
+auto [b,e] = ranges::search(subrange(it, ranges::end(@\exposid{base_}@)), @\exposid{pattern_}@);
+if (b != ranges::end(@\exposid{base_}@) && ranges::empty(@\exposid{pattern_}@)) {
+  ++b;
+  ++e;
+}
+return {b,e};
+\end{codeblock}
+\end{itemdescr}
+
+\rSec3[range.split.iterator]{Class \tcode{split_view::\exposid{iterator}}}
+
+\begin{codeblock}
+namespace std::ranges {
+  template<@\libconcept{forward_range}@ V, @\libconcept{forward_range}@ Pattern>
+    requires @\libconcept{view}@<V> && @\libconcept{view}@<Pattern> &&
+             @\libconcept{indirectly_comparable}@<iterator_t<V>, iterator_t<Pattern>, ranges::equal_to>
+  class split_view<V, Pattern>::@\exposid{iterator}@ {
+  private:
+    split_view* @\exposid{parent_}@ = nullptr;                              // \expos
+    iterator_t<V> @\exposid{cur_}@ = iterator_t<V>();                       // \expos
```

Comment:
@timsong-cpp, @BRevzin : is this still OK, or does this need to be changed to not have DMIs?

Reply:
@jwakely says it's correct as written, with the DMIs.

Reply:
Yes, this iterator is forward, so it's `regular` and hence always `default_initializable`.